tests/api-watch-diff/test.ml

```diff
@@ -1,4 +1,400 @@
+open Types
+open Predef
+
 let%expect_test "test_diff_interface" =
   let result = Api_watch_diff.diff_interface ~reference:[] ~current:[] in
   Format.printf "%b" result;
   [%expect {|false|}]
+
+let ref_signature =
+  (* Signature for ref.mli:
+     > type t = int
+     > type unused_type = string
+     > val f : t-> string *)
+  [
+    Sig_type
+      ( Ident.create_persistent "t",
+        {
+          type_params = [];
+          type_arity = 0;
+          type_kind = Type_abstract;
+          type_private = Public;
+          type_manifest = Some type_int;
+          type_variance = [];
+          type_separability = [];
+          type_is_newtype = false;
+          type_expansion_scope = 0;
+          type_loc = Location.none;
+          type_attributes = [];
+          type_immediate = Unknown;
+          type_unboxed_default = false;
+          type_uid = Uid.internal_not_actually_unique;
+        },
+        Trec_not,
+        Exported );
+    Sig_type
+      ( Ident.create_persistent "unused_type",
+        {
+          type_params = [];
+          type_arity = 0;
+          type_kind = Type_abstract;
+          type_private = Public;
+          type_manifest = Some type_string;
+          type_variance = [];
+          type_separability = [];
+          type_is_newtype = false;
+          type_expansion_scope = 0;
+          type_loc = Location.none;
+          type_attributes = [];
+          type_immediate = Unknown;
+          type_unboxed_default = false;
+          type_uid = Uid.internal_not_actually_unique;
+        },
+        Trec_not,
+        Exported );
+    Sig_value
+      ( Ident.create_persistent "f",
+        {
+          val_type =
+            Transient_expr.type_expr
+              (Transient_expr.create
+                 (Tarrow
+                    ( Nolabel,
+                      Transient_expr.type_expr
+                        (Transient_expr.create
+                           (Tconstr
+                              ( Path.Pident (Ident.create_predef "t"),
```

Comment:
I think predef is reserved for ocaml base types, e.g. types predefined by the compiler, such as `int`, `string`, `nativeint` or `bool`